### ISSUE TITLE
JoinLayers(): no applicable method for 'JoinLayers' applied to an object of class "c('SCTAssay', 'Assay', 'KeyMixin')" when use SCTintegration object

### ISSUE THREAD

**Issue by Jiahuan90** [satijalab/seurat#9211]
Hi, 

I encountered a situation, when I used SCTintegration seurat object, 
`#Merge_seurat <- IntegrateLayers(object = Merge_seurat, method = RPCAIntegration, normalization.method = "SCT", verbose = F)`
`Merge_seurat <- FindNeighbors(Merge_seurat, dims = 1:30, reduction = "integrated.dr")`
`Merge_seurat <- FindClusters(Merge_seurat, resolution = 0.6, cluster.name = "RPCA_clusters")`
`Merge_seurat <- RunUMAP(Merge_seurat, dims = 1:30, reduction = "integrated.dr")`
`Merge_seurat <- PrepSCTFindMarkers(Merge_seurat)`
`Merge_seurat <- JoinLayers(Merge_seurat, assay = 'SCT')`

if I want to join layers from Merge_seurat, it returned `JoinLayers(): no applicable method for 'JoinLayers' applied to an object of class "c('SCTAssay', 'Assay', 'KeyMixin')" when use SCTintegration object`. How can I solve the problem?

Thank you for youor kindly help~

**Comment by Riley-Grindle** [comment]
Also having a similar problem.

**Comment by MengYunWang** [comment]
The same issue here.

**Comment by afletch00** [comment]
Yup. Me to. I can't figure out how to turn these Seurat V5 objects into sce objects for the life of me.

**Comment by iddryg** [comment]
I'm experiencing the same issue.

**Comment by athulmenon** [comment]
Same issue here!

**Comment by ahwanpandey** [comment]
same, please help!

**Comment by ahwanpandey** [comment]
Hmm, seems to have worked if I use ```assay = "Spatial"```

```
merged_data.joined <- JoinLayers(merged_data, assay = "Spatial")
```

**Comment by avfentor** [comment]
same here, anybody found a solution yet? It's not worked for me even if I try to set either of my available assays

**Comment by jvelghe** [comment]
I also encountered this error, and would also appreciate some insight about layer differences between SCT and RNA assays.

**Comment by WilliamMWei** [comment]
> Hmm, seems to have worked if I use `assay = "Spatial"`
> 
> ```
> merged_data.joined <- JoinLayers(merged_data, assay = "Spatial")
> ```



> I also encountered this error, and would appreciate some insight about layer differences between SCT and RNA assays. I had to change back to RNA assay, join layers, then switch back to the SCT assay.
> 
> `DefaultAssay(obj) <- "RNA"` `obj <- JoinLayers(obj, assay = "RNA")`

I have similar issue while specifying assay = "SCT" or assay = "RNA" doesn't do anything..
Many thanks if your have any updates!

**Comment by kevinbrennan** [comment]
Same problem here

**Comment by sergio-rutella** [comment]
I set the DefaultAssay to "RNA" before joining layers; this solved the issue, apparently. I hope it helps.

**Comment by gevro** [comment]
Hi, Is this issue solved yet? Seems like a bug in how JoinLayers treats SCT (post SCTransform) assay data.

**Comment by Leonrunning** [comment]
> Hi, Is this issue solved yet? Seems like a bug in how JoinLayers treats SCT (post SCTransform) assay data.
> 
> Perhaps the answer is that JoinLayers should NOT be performed after split -> SCTransform -> IntegrateLayers, because PrepSCTFindMarkers and FindMarkers or FindAllMarkers should anyway be performed on the "SCT" layer rather than the "RNA" layer?

I met the same issue AND I aggre the SCT should be used for DEs. So, how to joint layers in SCT?

**Comment by dartagnan323** [comment]
Hi indeed same issue here. I have split assay by condition (= STIM in the tutorial) performed SCT then integration like in the tutorial. 
But then if I try to use FindConservedMarkers for one seurat cluster it asks me to join layers which I can't due to same error.  I performed PrepSCT already.
thanks for any help

**Comment by davidepernici** [comment]
No answers about this?? How are we supposed to continue downstream analysis if it's not possible to rejoin layers on SCT integration?

**Comment by jlmcalp** [comment]
Hi! Maybe I'm missing something, but it looks like the documentation for PrepSCTFindMarkers assumes that your Seurat object may be a merged object with multiple datasets. 

"Given a merged object with multiple SCT models, this function uses minimum of the median UMI (calculated using the raw UMI counts) of individual objects to reverse the individual SCT regression model using minimum of median UMI as the sequencing depth covariate. The counts slot of the SCT assay is replaced with recorrected counts and the data slot is replaced with log1p of recorrected counts."

https://rdrr.io/github/mrod0101/seurat/man/PrepSCTFindMarkers.html

**Comment by minimouse1** [comment]
See [Joining Layers after SCTransform](https://github.com/satijalab/seurat/issues/8558#top) #8558.  Apparently there are no layers in an SCT to join hence why JoinLayers fails.  Important point though if merging datasets: run SCTransform on the individual datasets first, then merge and run PrepSCTFindMarkers as shown in https://satijalab.org/seurat/reference/prepsctfindmarkers.  Running SCTransform on the entire merged dataset and then using PrepSCTFindMarkers doesn't work.
